addons/stock/models/stock_inventory.py

```diff
@@ -288,57 +223,66 @@ def _get_inventory_lines_values(self):
                 product_data['product_uom_id'] = Product.browse(product_data['product_id']).uom_id.id
                 quant_products |= Product.browse(product_data['product_id'])
             vals.append(product_data)
-        if self.exhausted:
-            exhausted_vals = self._get_exhausted_inventory_line(products_to_filter, quant_products)
-            vals.extend(exhausted_vals)
         return vals
 
-    def _get_exhausted_inventory_line(self, products, quant_products):
-        '''
-        This function return inventory lines for exausted products
-        :param products: products With Selected Filter.
-        :param quant_products: products available in stock_quants
-        '''
-        vals = []
-        exhausted_domain = [('type', 'not in', ('service', 'consu', 'digital'))]
-        if products:
-            exhausted_products = products - quant_products
-            exhausted_domain += [('id', 'in', exhausted_products.ids)]
-        else:
-            exhausted_domain += [('id', 'not in', quant_products.ids)]
-        exhausted_products = self.env['product.product'].search(exhausted_domain)
-        for product in exhausted_products:
-            vals.append({
-                'inventory_id': self.id,
-                'product_id': product.id,
-                'location_id': self.location_id.id,
-            })
-        return vals
 
 
 class InventoryLine(models.Model):
     _name = "stock.inventory.line"
     _description = "Inventory Line"
     _order = "product_id, inventory_id, location_id, prod_lot_id"
 
+    def _default_inventory_id(self):
+        return self._context.get('active_inventory', None)
+
+    def _domain_location_id(self):
+        domain = [('usage', 'in', ['internal', 'transit'])]
```

Comment:
It seems it doesn't work correctly when we re-open the inventory lines view.
I think the problem is, since the domain is set when trigger a onchange on the `inventory_id` field, once this field is set, as the field is stored, the onchange is triggered only the first time when the line is created.
So, the next times we open this view, as the field `inventory_id` is already set, we don't change/update it, so the onchange isn't triggered and the domain isn't defined.